app/src/NativeBase/Components/Project/ProjectRegisterInterest.tsx

```diff
@@ -0,0 +1,225 @@
+        setResponseHeader('Application Received')
+        setResponseMessage(
+          'Your request has been received. The STA team will respond shortly.',
+        )
+        navigation.setOptions({
+          title: 'Registration confirmed',
+        })
+      }
+      if (responseError) {
+        setResponseHeader('Something went wrong')
+        setResponseMessage(
+          "Sorry, we couldn't send your message - please try again. If this keeps happening, please contact the STA Volunteer App team.",
+        )
+      }
+    }
+  }, [responseData, responseError, navigation])
+
+  const onClose = () => {
+    setModalVisible(false)
+    goBack()
+  }
+  const validateField = (fieldName: string, value: string): boolean => {
+    let valid = true
+    let errorType = 'missing' as 'invalid' | 'missing'
+
+    switch (fieldName) {
+      case 'email':
+        if (!value) {
+          valid = false
+        } else if (!validateEmail(value)) {
+          valid = false
+          errorType = 'invalid'
+        }
+        break
+      default:
+        if (!value) {
+          valid = false
+        }
+        break
+    }
+
+    if (valid) {
+      setErrors(latestErrors => {
+        const updatedErrors = { ...latestErrors }
+        delete updatedErrors[fieldName]
+
+        return updatedErrors
+      })
+    } else {
+      setErrors(latestErrors => ({
+        ...latestErrors,
+        [fieldName]: {
+          type: errorType,
+        },
+      }))
+    }
+
+    return valid
+  }
+
+  const validateForm = (): boolean => {
+    const firstNameValid = validateField('firstName', firstName)
+    const lastNameValid = validateField('lastName', lastName)
+    const emailValid = validateField('email', email)
+
+    return firstNameValid && lastNameValid && emailValid
+  }
+
+  const submitForm = (): void => {
+    const formIsValid = validateForm()
+
+    if (formIsValid) {
+      setLoading(true)
+
+      registerInterest({
+        project: {
+          it_key: project.it_key,
+          res_id: project.res_id,
+        },
+        user: {
+          firstName,
+          lastName,
+          email,
+          lookingForPeerSupport,
+          availableFrom: dayjs(availableFromDate).format('YYYY-MM-DD'),
+        },
+      })
+    }
+  }
+  const { height } = Dimensions.get('window')
+  console.log(height)
+  return (
+    <>
+      <ResponseModal
+        isOpen={modalVisible}
+        header={responseHeader}
+        message={responseMessage}
+        success={success}
+        onClose={onClose}
+      />
+
+      <VStack minHeight={height - heightOfTopOfAppAndButton} marginBottom="10">
+        <TextInputControl
+          error={errors.hasOwnProperty('firstName')}
+          errorType={errors.firstName?.type}
+          label="First Name"
+          onBlur={() => validateField('firstName', firstName)}
+          onChange={setFirstName}
+          type="firstName"
+          value={firstName}
+          required={true}
```

Comment:
Optionally you could use the same shorthand in this file as mentioned earlier i.e. just writing `required` does the same job